os.go

```diff
@@ -189,42 +190,39 @@ func isHidden(f os.FileInfo, path string, hiddenfiles []string) bool {
 	hidden := false
 	for _, pattern := range hiddenfiles {
 		matched := matchPattern(strings.TrimPrefix(pattern, "!"), f.Name(), path)
-		if strings.HasPrefix(pattern, "!") && matched {
-			hidden = false
-		} else if matched {
-			hidden = true
+		if !matched {
+			continue
 		}
+		hidden = !strings.HasPrefix(pattern, "!")
 	}
 	return hidden
 }
 
 func userName(f os.FileInfo) string {
 	if stat, ok := f.Sys().(*syscall.Stat_t); ok {
-		uid := fmt.Sprint(stat.Uid)
+		uid := strconv.FormatUint(uint64(stat.Uid), 10)
```

Comment:
This should be faster, see https://stackoverflow.com/questions/24886015/how-to-convert-uint32-to-string#comment94954590_24886089